pkg/deploy/util/util.go

```diff
@@ -429,6 +433,52 @@ func DeploymentsForCleanup(configuration *deployapi.DeploymentConfig, deployment
 	return relevantDeployments
 }
 
+// WaitForRunningDeployerPod waits a given period of time until the deployer pod
+// for given replication controller is not running.
+func WaitForRunningDeployerPod(podClient kclient.PodsNamespacer, rc *api.ReplicationController, timeout time.Duration) (bool, error) {
+	podName := DeployerPodNameForDeployment(rc.Name)
+	if len(podName) == 0 {
+		return false, nil
+	}
+	isPodTerminated := func(p *api.Pod) bool {
+		return api.PodSucceeded == p.Status.Phase || api.PodFailed == p.Status.Phase
+	}
+	pod, err := podClient.Pods(rc.Namespace).Get(podName)
+	if err != nil {
+		return false, err
+	}
+	if isPodTerminated(pod) {
```

Comment:
You also need to return when it's running. Add it in your closure too and rename it to something like `canGetLogs`.